pagefetcher.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python2
+
+import urllib
+import os
+from __builtin__ import str
+
+# set of dictionaries to store url's and names
+CISCO = {"name1": "url1", "name2": "url2"}
+TEMPLATE_TO_URL = {"cisco": CISCO}
+
+
+class UrlNotAvailable(Exception):
+    """ Exception class to raise in case of a bad URL. """
+
+    def __init__(self):
+        Exception.__init__(self, "The given URL is not available!")
+
+
+class TemplateNotAvailable(Exception):
+    """ Exception class to raise in case of non-existence template. """
+
+    def __init__(self):
+        Exception.__init__(self, "The given template is not available!")
+
+
+class AurgumentIsNotAString(Exception):
+    """ Exception class to raise in case of non-string input. """
+
+    def __init__(self):
+        Exception.__init__(self, "The given argument is not a string!")
+
+
+def fetch(template):
+    """
+    Make a new folder with the name provided in template and fetch all the
+    files for the given template.
+
+    Args:
+        template (str): The name of requested template.
+
+    Returns:
+        None
+
+    Raises:
+        TemplateNotAvailble: If the given template is not in the database.
+        AurgumentIsNotAString: if the given template is not a string.
+    """
+
+    # check to see if template is a string
+    if not type(template) is str:
+        raise AurgumentIsNotAString()
+
+    # remove the directory and get the name of the template
+    user_choice = template[template.find("/")+1:]
```

Comment:
Instead of `user_choice`, I recommend something more specific, such as `chosen_template`.